chore: Spotless formatter and check

MarcinVaadin opened the pull request:
As in title

MarcinVaadin committed:
Spotless formatter and check

MarcinVaadin commented:
Eclipse formatter is adding space in JCP comments `// #endif` which is wrong in this case.

Don't know why but setting:
```
<setting id="org.eclipse.jdt.core.formatter.format_line_comment_starting_on_first_column" value="false"/>
<setting id="org.eclipse.jdt.core.formatter.comment.format_line_comments" value="false"/>
```

does not affect anything

MarcinVaadin committed:
Merge branch 'v24.8' into chore/spotless

peholmst committed:
Merge branch 'v24.8' into chore/spotless

peholmst committed:
Fix broken //# comments

peholmst committed:
Fix more broken //# comments

peholmst committed:
Fix more broken //# comments

peholmst committed:
Update Spotless and Eclipse JDT versions

peholmst commented:
It looks like the `walking-skeleton/eclipse-formatter.xml` rules are being used for that module even when the goal is invoked from the parent with `/eclipse-formatter.xml`. The skeleton rules have line formatting enabled. Disabling line formatting in that file has the desired effect.

peholmst committed:
Disable line formatting

peholmst committed:
Reformatted by Spotless

peholmst committed:
Use 120 characters for line length (same as Spring)

peholmst committed:
Use the skeleton's own formatter rules

It appears they override the rules in the root project anyway. No need to keep double files that are identical.

peholmst committed:
Reformat file

MarcinVaadin commented on eclipse.importorder:
```diff
@@ -0,0 +1,6 @@
+2=java
+4=com.google.gwt
+5=com.vaadin
+6=elemental
```
We switched to `com.fasterxml.jackson` in meantime, maybe here should also be updated.

peholmst committed:
Replace elemental with com.fasterxml.jackson

peholmst reviewed (approved):
Marking as approved even though I have made several of the changes myself.
